docs/core/testing/microsoft-testing-platform-migration-from-v1-to-v2.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Microsoft.Testing.Platform migration from v1 to v2
+description: Learn about migrating to Microsoft.Testing.Platform v2.
+author: Youssef1313
+ms.author: ygerges
+ms.date: 09/23/2025
+---
+
+# Migrate from Microsoft.Testing.Platform v1 to Microsoft.Testing.Platform v2
+
+The preview version Microsoft.Testing.Platform v2 is now available. This migration guide explores what's changed in Microsoft.Testing.Platform v2 and how you can migrate to this version.
+
+## Source breaking changes
+
+These are breaking changes that might cause build errors after migration.
+
+### Dropped unsupported target frameworks
+
+Support for target frameworks .NET Core 3.1 to .NET 7 is dropped. The minimum supported .NET version is .NET 8.
+This change doesn't affect .NET Framework. .NET Framework 4.6.2 continues to be the minimum supported .NET Framework target.
+
+### Remove obsolete types
+
+The following types were marked obsolete in MTP v1. In MTP v2, they are now removed:
+
+- `ITestApplicationLifecycleCallbacks`: use `ITestHostApplicationLifetime` instead.
+- `TestNodeFileArtifact`: use `FileArtifactProperty` instead.
+- `KeyValuePairStringProperty`: use `TestMetadataProperty` instead.
+- `TestSessionContext.Client`: use `IClientInfo` instead.
+
+This breaking change doesn't affect typical users of test frameworks. It only affects test framework authors and extension authors.
+
+### API signature changes
+
+- A `CancellationToken` parameter was added to `IOutputDevice.DisplayAsync`.
```

Comment:
Use an xref for IOutputDevice.DisplayAsync?? There's a [VS Code extension](https://github.com/dotnet/docs-tools/blob/main/xref-helper/README.md) you can install to help with xrefs.